pkg/test/step.go

```diff
@@ -533,6 +551,42 @@ func (s *Step) LoadYAML(file string) error {
 			} else {
 				return fmt.Errorf("failed to load TestAssert object from %s: it contains an object of type %T", file, obj)
 			}
+
+			var errs []error
+			for _, resourceRef := range s.Assert.ResourceRefs {
+				if err := resourceRef.Validate(); err != nil {
+					errs = append(errs, fmt.Errorf("validation failed for reference '%v': %w", resourceRef.String(), err))
+				}
+			}
+
+			if len(errs) > 0 {
+				return fmt.Errorf("failed to load TestAssert object from %s: %w", file, errors.Join(errs...))
+			}
+
+			var assertions []*harness.Assertion
+			assertions = append(assertions, s.Assert.AssertAny...)
+			assertions = append(assertions, s.Assert.AssertAll...)
+
+			env, err := testutils.BuildEnv(s.Assert.ResourceRefs)
+			if err != nil {
+				return fmt.Errorf("failed to load TestAssert object from %s: %w", file, err)
+			}
+
+			if len(assertions) > 0 {
+				s.Programs = make(map[string]cel.Program)
+			}
+
+			for _, assertion := range assertions {
+				if prg, err := assertion.BuildProgram(env); err != nil {
+					errs = append(errs, err)
+				} else {
+					s.Programs[assertion.CELExpression] = prg
+				}
+			}
+
+			if len(errs) > 0 {
+				return fmt.Errorf("failed to load TestAssert object from %s: %w", file, errors.Join(errs...))
+			}
```

Comment:
Please extract this code into a `func loadPrograms(*harness.TestAssert) (map[string]cel.Program, error)` or similar and call it here.